uPortSDK/Classes/Crypto/Crypto.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  Crypto.swift
+//  uPortSDK
+//
+
+import Foundation
+import Sodium
+import Clibsodium
+
+/**
+ * This struct exposes methods to encrypt and decrypt messages according to the uPort spec at
+ * https://github.com/uport-project/specs/blob/develop/messages/encryption.md
+ */
+public struct Crypto
+{
+    public struct Constants
+    {
+        static let asyncEncryptionAlgorithm = "x25519-xsalsa20-poly1305"
+        static let blockSize = 64.0
+    }
+    /**
+     * This class encapsulates an encrypted message that was produced using
+     * https://github.com/uport-project/specs/blob/develop/messages/encryption.md
+     */
+    public struct EncryptedMessage: Codable
+    {
+        var version: String = Constants.asyncEncryptionAlgorithm
+        var nonce: String
+        var ephemPublicKey: String
+        var ciphertext: String
+        
+        
+        public init(nonce: String, ephemPublicKey: String, ciphertext: String)
+        {
+            self.nonce = nonce
+            self.ephemPublicKey = ephemPublicKey
+            self.ciphertext = ciphertext
+        }
+        
+        public func toJson() -> Data
+        {
+            return try! Data(JSONEncoder().encode(self))
+        }
+        
+        public static func fromJson(jsonData: Data) -> EncryptedMessage
+        {
+            return try! JSONDecoder().decode(EncryptedMessage.self, from: jsonData)
```

Comment:
This can `throw` an exception (when `jsonData` is invalid). With `try!` the app will crash, which is not acceptable of course. Best would be to make this function `throws`.